libraries/Microsoft.Bot.Connector/Authentication/SimpleServiceClientCredentialFactory.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Rest;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    /// <summary>
+    /// A simple implementation of the <see cref="IServiceClientCredentialsFactory"/> interface.
+    /// </summary>
+    public class SimpleServiceClientCredentialFactory : IServiceClientCredentialsFactory
+    {
+        private HttpClient _httpClient;
+
+        private ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SimpleServiceClientCredentialFactory"/> class.
+        /// with empty credentials.
+        /// </summary>
+        public SimpleServiceClientCredentialFactory()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SimpleServiceClientCredentialFactory"/> class.
+        /// with the provided credentials.
+        /// </summary>
+        /// <param name="appId">The app ID.</param>
+        /// <param name="password">The app password.</param>
+        /// <param name="httpClient">A custom httpClient to use.</param>
+        /// <param name="logger">A logger instance to use.</param>
+        public SimpleServiceClientCredentialFactory(string appId, string password, HttpClient httpClient, ILogger logger)
+        {
+            AppId = appId;
+            Password = password;
+            _httpClient = httpClient;
```

Comment:
Here we are not defaulting httpclient and logger to null. What's the contract? Do we allow null? Do we default?

Reply:
i think your right about the logger - it should be logger.null or something - and I think we defer on httpclient i.e. allow it to be null here and then check later